### PR TITLE
Add support for debugging macOS app bundles

### PR DESCRIPTION
**Pull request by WardenGnaw**
This PR adds support for debugging app bundles on macOS.

LLDB understands how to handle app bundles when provided to
-file-exec-and-symbols.

Reference:
./lldb-mi
(gdb)
-file-exec-and-symbols /Applications/VLC.app
^done
(gdb)
=library-loaded,id="/Applications/VLC.app/Contents/MacOS/VLC",target-name="/Applications/VLC.app/Contents/MacOS/VLC",host-name="/Applications/VLC.app/Contents/MacOS/VLC",symbols-loaded="0",loaded_addr="-",size="20480"

Added MacOSApp Test to CppTests.

### CODE REVIEW

**Comment by gregg-miskelly** [comment]
LGTM

**Comment by WardenGnaw** [comment]
This is pending testing on macOS :)

**Comment by gregg-miskelly** [comment]
> @gregg-miskelly Perchance do you know how to default each test to have this new compiler as unsupported or do I have to add an UnsupportedCompiler tag to the existing tests?

I have no idea. You could ask the Production Diag folks (Justin Anderson, Del) if they remember anything.

**Comment by WardenGnaw** [comment]
@gregg-miskelly When you have the time, can you re-review this PR. I was able to add a XCode project that compiles a `.app` and we test a simple debug launch against it.

File: src/OpenDebugAD7/AD7DebugSession.cs [hunk 1]
```diff
@@ -171,6 +171,14 @@ private bool ValidateProgramPath(ref string program)
             program = m_pathConverter.ConvertLaunchPathForVsCode(program);
             if (!File.Exists(program))
             {
+                // On macOS, check to see if we are trying to debug an app bundle (.app).
+                // 'app bundles' contain various resources and executables in a folder.
+                // LLDB understands how to target these bundles.
+                if (Utilities.IsOSX() && program.EndsWith(".app", StringComparison.OrdinalIgnoreCase) && miMode.Equals("lldb", StringComparison.OrdinalIgnoreCase))
```
Comment (gregg-miskelly) [inline src/OpenDebugAD7/AD7DebugSession.cs:177]: >. [](http://example.com/codeflow?start=111&length=1)

`?.Equals(...) == true`

**Review by gregg-miskelly — APPROVED** [review]
Otherwise LGTM

**Pull request merged**